### ISSUE TITLE
Feature request: Option to just pull comments

### ISSUE THREAD

**Issue by Copyrites** [instaloader/instaloader#131]
As far as I can tell, if I use instaloader to grab the comments on a post, it will also download the picture too.   My goal is to track comments, in particular I want to log comments before they are deleted.  So that means I have to query the servers frequently, and re-downloading the picture each time would put a unnecessary burden on instagram's servers.

I'm thinking that either a  --comments-only option or a --no-pictures option would do the trick.

Also, thanks for this great tool, its already been very helpful.

**Comment by SilverFuzz** [comment]
Would be much appreciated. I use instaloader for law enforcement and it is of utmost importance that I can download all comments of a specific post with all available metadata. 
Right now it doesn't work at all for posts with many comments (e.g. 50k). I get the 429 problem at once without a single comment downloaded.

**Comment by aandergr** [comment]
We plan to somehow implement a `--no-picture` in the upcoming version 4.1. However, it is not trivial, since Instaloader's `--fast-update` uses presence of jpg files as indicator of whether a picture has already been downloaded or not. The *commit mode* suggested in #139 might also have implications regarding this.

As a **"quick fix"**, in `instaloader/instaloader.py`, comment out lines 344 to 363 (the lines in `Instaloader.download_post()` that download the picture). However, it will break `--fast-update`.

You could also use Instaloader as a module in a script that goes through a profile's post (`Profile.get_posts()`) and calls `Instaloader.update_comments()` on them, just as `Instaloader.download_post()` does, simply without downloading the post. The `instaloader` Python module documentation is here: https://instaloader.github.io/as-module.html

> So that means I have to query the servers frequently, and re-downloading the picture each time would put a unnecessary burden on instagram's servers.

Instaloader does no extra query for the JPG URL, since it is included in the post metadata structure which already has to be queried to obtain comments. Hence the only unnecessary burden (except of your traffic) is the download of the JPG itself, which are hosted in a simple (and efficient) CDN. Also, the download time should be minimal compared with how long it takes to iterate through all comments within Instagram's rate limits. In my opinion, unless there are bandwith or traffic constraints on your side, there is no need to worry about the burden.

> Right now it doesn't work at all for posts with many comments (e.g. 50k).

This has been reported in issue #125. However, we only can access information that Instagram provides us. Is there an app that can access all comments? If not, chances are low that we can fix this problem.

> There is a severe limitation to 200 requests per hour in the API, but right now instaloader gives a 429 error right away.

Instaloader should not give a 429 if used properly (which includes mass picture/metadata mining, since that's what Instaloader should be designed for). If you get a 429 right away, that's a bug. We ask you to report it as such, and provide us information about when it happens. (Most probably obvious, but we collected a few points to keep in mind when reporting a bug here: https://instaloader.github.io/contributing.html#reporting-bugs)

Instaloader does not use the API, so the *200 request per hour* do not apply for us. We use the browser interface which has a different rate limit, which we currently believe to be 20 requests per 11 minutes, which is about 5000 comments per hour.

A general note about **429 - Too Many Requests**:
> Instaloader has a logic to keep track of its requests to Instagram and to obey their rate limits. Since they are nowhere documented, we try them out experimentally. We have a daily cron job running to confirm that Instaloader still stays within the rate limits. Nevertheless, the rate control logic assumes that
> - at one time, Instaloader is the only application that consumes requests. I.e. neither the Instagram browser interface, nor a mobile app, nor another Instaloader instance is running in parallel,
> - no requests had been consumed when Instaloader starts.
>
> The latter one implies that restarting or reinstantiating Instaloader often within short time is prone to cause a 429. When a request is denied with a 429, Instaloader retries the request as soon as the temporary ban is assumed to be expired.

Best regards
Alex

**Comment by aandergr** [comment]
Implemented in `v4.1-dev`.

```
pip install --upgrade git+https://github.com/instaloader/instaloader@v4.1-dev
```
